src/track/keyfactory.cpp

```diff
@@ -106,7 +106,14 @@ Keys KeyFactory::makePreferredKeys(
             pChange->set_key(it->first);
             pChange->set_frame_position(static_cast<int>(frame));
         }
-        key_map.set_global_key(KeyUtils::calculateGlobalKey(key_changes, totalFrames, sampleRate));
+
+        mixxx::track::io::key::ChromaticKey global_key =
+                KeyUtils::calculateGlobalKey(
+                        key_changes, totalFrames, sampleRate);
+        key_map.set_global_key(global_key);
+        QString global_key_text = KeyUtils::keyToString(
+                global_key, KeyUtils::KeyNotation::ID3v2);
```

Comment:
what implications does hardcoding the notation have? Does this overwrite the user-notation?

Reply:
Not affected. This text is generated from the underlying data dynamically according to the preferences.

Reply:
This ensures that we write the ID3v2 to the tags. https://github.com/mixxxdj/mixxx/issues/10890

Reply:
Great. thanks for the clarification.